src/smc-webapp/account/init.ts

```diff
@@ -0,0 +1,86 @@
+import * as misc from "smc-util/misc";
+const { webapp_client } = require("../webapp_client");
+const remember_me = webapp_client.remember_me_key();
+import { AccountActions } from "./actions";
+import { AccountStore } from "./store";
+import { AccountTable } from "./table";
+
+export function init(redux) {
+  // Register account store
+  // Use the database defaults for all account info until this gets set after they login
+  const init = misc.deep_copy(
+    require("smc-util/schema").SCHEMA.accounts.user_query.get.fields
+  );
+  // ... except for show_global_info2 (null or a timestamp)
+  init.other_settings.show_global_info2 = "loading"; // indicates there is no data yet
+  init.editor_settings.physical_keyboard = "NO_DATA"; // indicator that there is no data
+  init.user_type = misc.get_local_storage(remember_me)
+    ? "signing_in"
+    : "public"; // default
+  const store = redux.createStore("account", AccountStore, init);
+  const actions = redux.createActions("account", AccountActions);
+
+  actions._init(store);
+
+  redux.createTable("account", AccountTable);
+
+  // Login status
+  webapp_client.on("signed_in", function(mesg) {
+    if (mesg != null ? mesg.api_key : undefined) {
+      // wait for sign in to finish and cookie to get set, then redirect
+      const f = () =>
+        (window.location.href = `https://authenticated?api_key=${mesg.api_key}`);
+      setTimeout(f, 2000);
```

Comment:
Well this code (that I wrote) is scary!  It would probably randomly fail on slow computers...  Yikes. Not your fault or something for you to do anything about.